libs/utils/utils.js

```diff
@@ -568,3 +501,218 @@ export function createIntersectionObserver({ el, callback, once = true, options
   io.observe(el);
   return io;
 }
+
+export const getLocaleIetf = () => getConfig().locale?.ietf;
+
+/**
+ * Returns the language dependent root path
+ * @returns {string} The computed root path
+ */
+export const getPrefix = () => getConfig().locale?.prefix;
+
+export const emptyDiv = () => document.createElement('div');
+
+/**
+ * fetches the string variables.
+ * @returns {object} localized variables
+ */
+
+export async function fetchPlaceholders() {
+  if (!window.placeholders) {
+    const resp = await fetch(`/${getPrefix()}/placeholders.json`);
+    const json = await resp.json();
+    window.placeholders = {};
+    json.data.forEach((placeholder) => {
+      window.placeholders[placeholder.Key] = placeholder.Text;
+    });
+  }
+  return window.placeholders;
+}
+
+export function stamp(message) {
+  if (window.name.includes('performance')) {
+    // eslint-disable-next-line no-console
+    console.warn(`${new Date() - performance.timeOrigin}:${message}`);
+  }
+}
+
+/*
+ *
+ * Taxonomy Utils
+ *
+*/
+
+let taxonomyModule;
+
+export function getTaxonomyModule() {
+  return taxonomyModule;
+}
+
+/**
+ * For the given list of topics, returns the corresponding computed taxonomy:
+ * - category: main topic
+ * - topics: tags as an array
+ * - visibleTopics: list of visible topics, including parents
+ * - allTopics: list of all topics, including parents
+ * @param {Array} topics List of topics
+ * @returns {Object} Taxonomy object
+ */
+function computeTaxonomyFromTopics(taxonomy, topics, path) {
```

Comment:
This should be moved out of utils. We only put things in utils if a large majority of a site uses a feature.